src/app.js

```diff
@@ -0,0 +1,79 @@
+import axios from "axios";
+
+
+// Referenties naar elementen in HTML document toewijzen aan variabelen
+const countryDetailsField = document.getElementById("country-details-field");
+const countryNameInput = document.getElementById("country-name-input-field");
+const countryNameInputForm = document.getElementById("country-name-input-form");
+const errorMessage = document.getElementById("error-message");
+
+
+// URI aan (final) variabele toewijzen
+// Lege variabele declareren om later de endpoint aan toe te kunnen wijzen
+const URI = "https://restcountries.com/";
+let endpoint = null;
+
+
+// Functie declareren die de entry (of entries) uit een array met currencies als één string teruggeeft
+function getCurrencies(listOfCurrencies) {
+    if (listOfCurrencies.length === 1) {
+        return listOfCurrencies[0].name;
+    } else {
+        return `${listOfCurrencies[0].name} and ${listOfCurrencies[1].name}`
+    }
+}
+
+
+// Asynchrone functie declareren om de data van een opgegeven land uit de API op te halen
+// De endpoint als variabele opnemen, deze waarde wordt later met de functie/event listener toegewezen
+// Variabele initaliseren om de opgehaalde data (response) aan toe te wijzen
+// Catch/try block (succes-scenario en error-afvang) toevoegen
+// Await key-word toevoegen zodat de gevraagde data eerst met Axios kan worden opgehaald voordat het script verder laadt
+// Eerder gedeclareerde functie getCurrencies aanroepen om de valuta van het betreffende land als string terug te geven en deze string aan een nieuwe variabele toewijzen
+// Alle specifieke opgehaalde data via de innerHTML property op de juiste plek in het HTML bestand invoegen
+
+async function fetchCountryDetails() {
+    try {
+        const response = await axios.get(URI + endpoint);
+        const {data: country} = response;
```

Comment:
Hier zou je nog verder spreading kunen toepassen
```suggestion
        const {data: countries } = response;
        const { flag, name, population, subregion, capital } = countries[0];
```